pkg/tests/end_to_end_tests/metric_ingest_bench_test.go

```diff
@@ -79,25 +79,65 @@ func BenchmarkMetricIngest(b *testing.B) {
 }
 
 func BenchmarkNewSeriesIngestion(b *testing.B) {
-	seriesGen, err := testsupport.NewSeriesGenerator(10, 100, 4)
-	require.NoError(b, err)
-
-	ts := seriesGen.GetTimeseries()
+	benchmarks := []struct {
+		name               string
+		numMetrics         int
+		numSeriesPerMetric int
+		numLabelsPerMetric int
+		batchSize          int
+	}{
+		{
+			name:               "small series",
+			numMetrics:         100,
+			numSeriesPerMetric: 10,
+			numLabelsPerMetric: 4,
+			batchSize:          100,
+		},
+		{
+			name:               "medium series",
+			numMetrics:         100,
+			numSeriesPerMetric: 100,
+			numLabelsPerMetric: 4,
+			batchSize:          1000,
+		},
+		{
+			name:               "large series",
+			numMetrics:         100,
+			numSeriesPerMetric: 1000,
+			numLabelsPerMetric: 4,
+			batchSize:          10000,
+		},
+	}
 
-	withDB(b, "bench_e2e_new_series_ingest", func(db *pgxpool.Pool, t testing.TB) {
-		metricsIngestor, err := ingestor.NewPgxIngestorForTests(pgxconn.NewPgxConn(db), &ingestor.Cfg{
-			NumCopiers:              8,
-			InvertedLabelsCacheSize: cache.DefaultConfig.InvertedLabelsCacheSize,
+	for _, benchmark := range benchmarks {
+		benchName := fmt.Sprintf("%s metrics %d seriesPerMetric %d labels %d batchSize %d", benchmark.name, benchmark.numMetrics, benchmark.numSeriesPerMetric, benchmark.numLabelsPerMetric, benchmark.batchSize)
+		b.Run(benchName, func(b *testing.B) {
+			seriesGen, err := testsupport.NewSeriesGenerator(benchmark.numMetrics, benchmark.numSeriesPerMetric, benchmark.numLabelsPerMetric)
+			require.NoError(b, err)
+
+			ts := seriesGen.GetTimeseriesInBatch(benchmark.batchSize)
+
+			for i := 0; i < b.N; i++ {
```

Comment:
I wonder if you have idea how consistent are the numbers if we run without `b.N` iterations? 
I am aiming at minimizing benchmark time if possible but still getting meaningful numbers...not sure if possible though...

Reply:
@niksajakovljevic I just kept the b.N convention here, but it runs only once 100% of the time because the block being  benchmarked always takes more than 1 second.